node/node.go

```diff
@@ -249,6 +250,17 @@ func (n *Node) CreateLedgerChannel(Counterparty types.Address, ChallengeDuration
 		// Appdata implicitly zero
 	)
 
+	// Check store to see if there is an existing channel with this counterparty
+	channelExists, err := directfund.ChannelsExistWithCounterparty(Counterparty, n.store.GetChannelsByParticipant, n.store.GetConsensusChannel)
+	if err != nil {
+		n.logger.Err(err).Msg("directfund.ChannelsExistWithCounterparty")
+		return directfund.ObjectiveResponse{}, fmt.Errorf("counterparty check failed: %w", err)
+	}
+	if channelExists {
+		n.logger.Error().Msg("directfund: channel already exists")
+		return directfund.ObjectiveResponse{}, fmt.Errorf("counterparty %s: %w", Counterparty, directfund.ErrLedgerChannelExists)
```

Comment:
Right now the rpc server will not recognize this error, so it will default to returning a generic `InternalServerError` (analogous to http status 500). Perhaps we could add a transformation step that converts `directfund.ErrLedgerChannelExists` error into a `InvalidParamsError` JsonRpcError, which could preserve the error message while adding the appropriate jsonrpc code to return to the client?

Reply:
I changed some code in `rpc/server.go` to preserve the error message in the response to the client. We will still default to use the `InternalServerError` code in the response. I feel like this type of error should be analogous to an http status 400 error but I'm not sure if `InvalidParamsError` is the correct one for that